src/emeus-constraint-layout.c

```diff
@@ -1407,7 +1407,12 @@ create_child_constraint (EmeusConstraintLayout      *layout,
   /* Alternatively, if it's not a constant value, we find the variable
    * associated with it
    */
-  if (constraint->source_object != NULL)
+  if (constraint->source_object == NULL ||
+      (EMEUS_IS_CONSTRAINT_LAYOUT (constraint->source_object) && constraint->source_object == layout))
```

Comment:
Actually, I think this is really redundant. It can be simplified with:

```cpp
  if (constraint->source_object == NULL || constraint->source_object == layout)
```